src/lib/logical_query_plan/projection_node.cpp

```diff
@@ -29,6 +29,35 @@ bool ProjectionNode::is_column_nullable(const ColumnID column_id) const {
   return node_expressions[column_id]->is_nullable_on_lqp(*left_input());
 }
 
+std::shared_ptr<LQPUniqueConstraints> ProjectionNode::unique_constraints() const {
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+  unique_constraints->reserve(node_expressions.size());
+
+  // Forward unique constraints, if applicable
+  const auto& input_unique_constraints = left_input()->unique_constraints();
+
+  const auto& expressions = this->output_expressions();
+  const auto output_expressions = ExpressionUnorderedSet{expressions.cbegin(), expressions.cend()};
+
+  for (const auto& input_unique_constraint : *input_unique_constraints) {
+    if (!has_output_expressions(input_unique_constraint.expressions)) {
+      // Future Work: Save unique constraint for next block (derived constraints, see comment down below)
```

Comment:
I wouldn't suggest code structure for code that has not yet been written yet. Simply move the comment from below here.